src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -0,0 +1,718 @@
+                    return;
+                }
+                double fullBucketRate = processBucketAggregationResults(aggregate);
+                if (fullBucketRate < CONFIG_BUCKET_MINIMUM_SUCCESS_RATE) {
+                    multiFeatureQueriesResponseListener
+                        .onFailure(
+                            new ADValidationException(
+                                CommonErrorMessages.FEATURE_QUERY_TOO_SPARSE + feature.getName(),
+                                DetectorValidationIssueType.FEATURE_ATTRIBUTES,
+                                ValidationAspect.MODEL
+                            )
+                        );
+                } else {
+                    multiFeatureQueriesResponseListener
+                        .onResponse(new MergeableList<>(new ArrayList<>(Collections.singletonList(new double[] { fullBucketRate }))));
+                }
+            }, e -> {
+                String errorMessage;
+                if (isExceptionCausedByInvalidQuery(e)) {
+                    errorMessage = CommonErrorMessages.FEATURE_WITH_INVALID_QUERY_MSG + feature.getName();
+                } else {
+                    errorMessage = CommonErrorMessages.UNKNOWN_SEARCH_QUERY_EXCEPTION_MSG + feature.getName();
+                }
+                logger.error(errorMessage, e);
+                multiFeatureQueriesResponseListener.onFailure(new OpenSearchStatusException(errorMessage, RestStatus.BAD_REQUEST, e));
+            }));
+        }
+    }
+
+    private void windowDelayRecommendation(long latestTime) {
+        long delayMillis = timeConfigToMilliSec(anomalyDetector.getWindowDelay());
+        if ((Instant.now().toEpochMilli() - latestTime > delayMillis)) {
+            long minutesSinceLastStamp = TimeUnit.MILLISECONDS.toMinutes(Instant.now().toEpochMilli() - latestTime);
+            listener
+                .onFailure(
+                    new ADValidationException(
+                        CommonErrorMessages.WINDOW_DELAY_REC + minutesSinceLastStamp,
+                        DetectorValidationIssueType.WINDOW_DELAY,
+                        ValidationAspect.MODEL,
+                        new IntervalTimeConfiguration(minutesSinceLastStamp, ChronoUnit.MINUTES)
+                    )
+                );
+            return;
+        }
+        listener.onResponse(null);
+    }
+
+    private LongBounds getTimeRangeBounds(long endMillis, IntervalTimeConfiguration detectorIntervalInMinutes) {
+        Long detectorInterval = timeConfigToMilliSec(detectorIntervalInMinutes);
+        Long startMillis = endMillis - ((long) getNumberOfSamples() * detectorInterval);
+        return new LongBounds(startMillis, endMillis);
+    }
+
+    private int getNumberOfSamples() {
+        long interval = anomalyDetector.getDetectorIntervalInMilliseconds();
+        return Math
+            .max(
+                (int) (Duration.ofHours(AnomalyDetectorSettings.TRAIN_SAMPLE_TIME_RANGE_IN_HOURS).toMillis() / interval),
```

Comment:
May need to handle divide by 0 error if the request is coming via API. Frontend prevents a value <=0 to be set here, but a direct request could cause a problem.

Reply:
Added this check on the backend as well: https://github.com/opensearch-project/anomaly-detection/blob/30d0636ccacfbfc5aa1844b66b35da2ba44f0d79/src/main/java/org/opensearch/ad/model/AnomalyDetector.java#L213